template_example_test.go

```diff
@@ -150,30 +133,20 @@ func lexCode(_ context.Context, l *lexparse.Lexer) (lexparse.LexState, error) {
 			l.Emit(lexTypeBlockEnd)
 			return lexparse.LexStateFn(lexText), nil
 		default:
-			rn, err := l.Peek(1)
-			// TODO(#95): update EOF check.
-			if errors.Is(err, io.EOF) {
-				break
-			}
-			if err != nil {
-				return nil, fmt.Errorf("lexing code: %w", err)
-			}
-			if unicode.IsSpace(rn[0]) {
+			if unicode.IsSpace(l.Peek()) {
```

Comment:
This won't work for things like `{{myvar}}` since it's looking for a trailing space after the variable identifier. It should check for runes that aren't identifier characters.